server/src/main/java/com/google/musicanalysis/api/perspective/PerspectiveRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.google.musicanalysis.site;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+
+public class PerspectiveRequest {
+
+  private String text;
+  private ArrayList<String> attributes;
+
+  public PerspectiveRequest(String text, ArrayList<String> attributes) {
+    this.text = text;
+    this.attributes = attributes;
+  }
+
+  /**
+   * Sends out request JSON Object and returns the JSON response from Perspective API.
+   *
+   * @return the JSON response as string
+   */
+  public String getResponse() throws MalformedURLException, IOException {
+
+    String jsonString = buildJson(this.text, this.attributes);
+    URL url = new URL("https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=");
+
+    // Open up the connection
+    HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    con.setRequestMethod("POST");
+    con.setRequestProperty("Content-Type", "application/json; utf-8");
+    con.setRequestProperty("Accept", "application/json");
+    con.setDoOutput(true);
+
+    // Send out the json string
+    try (OutputStream out = con.getOutputStream()) {
+      byte[] input = jsonString.getBytes("utf-8");
+      out.write(input, 0, input.length);
+    }
+
+    BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream(), "utf-8"));
+
+    StringBuffer response = new StringBuffer();
+
+    String chunk;
+    while ((chunk = in.readLine()) != null) {
+      response.append(chunk.trim());
+    }
+
+    in.close();
+
+    return response.toString();
+  }
+
+  /**
+   * Builds the JSON Object to be sent out.
+   *
+   * @param textToAnalyze The text that will be analyzed by the Perspective API.
+   * @param requestedAttributes the attributes requested.
+   */
+  private String buildJson(String textToAnalyze, ArrayList<String> requestedAttributes) {
+
```

Comment:
why are there 2 `buildJson()` methods in `PerspectiveRequest.java` and `PerspectiveJsonBuilder.java`. It's a bit confusing. Could you maybe distinguish them with diff names? One seems to be used as a helper within `PerspectiveRequest` and the other seems to be used as a jsonBuilder to be sent out. Maybe call the json builder to be sent out to be called `buildOutputJson()` (or whatever is suitable for your code)?

Reply:
Will do.

Reply:
I actually ended up deleting the `PerspectiveJsonBuilder.java` file completely in another pr so this issue has been resolved.

Reply:
I actually removed the whole file and renamed it to `buildFinalJson()`
Thank you for the suggestion.